source/solvers/gls_navier_stokes.cc

```diff
@@ -137,7 +138,12 @@ GLSNavierStokesSolver<dim>::setup_dofs_fd()
                        dsp,
                        this->mpi_communicator);
 
-  if (this->simulation_parameters.post_processing.calculate_average_velocities)
+  if (this->simulation_parameters.post_processing
+        .calculate_average_velocities ||
+      this->simulation_parameters.initial_condition->type ==
+        Parameters::InitialConditionType::average_velocity_profile ||
+      this->simulation_parameters.initial_condition->type ==
+        Parameters::InitialConditionType::average_velocity_profile)
```

Comment:
You have a repeated || statement here.
```suggestion
        .calculate_average_velocities ||
      this->simulation_parameters.initial_condition->type ==
        Parameters::InitialConditionType::average_velocity_profile)
```